app/components/buttons/regular_button_component/regular_button_component.html.haml

```diff
@@ -0,0 +1,36 @@
+- slim_class = @size == "slim" ? "slim " : " "
+- danger_class = @color == "danger" ? "danger-button " : " "
+- warning_class = @color == "warning" ? "warning-button " : " "
+- disabled_class = @state == "disabled" ? "disabled-button " : " "
+.button-container
+    - if @type == "link"
+        %a.regular-button{:name => @name, onclick: "displayAnimation()", id: "button-animation-id", class: "#{@variant}-button "+danger_class+warning_class+disabled_class+slim_class, href: @href}
+            - if @icon != "" && @icon_type == "left"
+                = inline_svg_tag @icon, class: "#{@variant}-button-icon left-button-icon"
+            = @value
+            - if @icon != "" && @icon_type == "right"
+                = inline_svg_tag @icon, class: "#{@variant}-button-icon right-button-icon"
+    - else
```

Comment:
instead of `if` and `else` here, create two components, one for buttons and another one that extends it for links.